kolibri/plugins/safe_html5_viewer/assets/src/views/Lightbox.vue

```diff
@@ -0,0 +1,398 @@
+        const maxH = this.backdropSize.height - (isSmallWindow ? 32 : 64);
+
+        const img = this.$refs.imageRef;
+        if (!img) return;
+        const naturalW = img.naturalWidth;
+        const naturalH = img.naturalHeight;
+
+        const widthRatio = maxW / naturalW;
+        const heightRatio = maxH / naturalH;
+        const scale = Math.min(widthRatio, heightRatio, 1);
+        this.baseSize.width = Math.round(naturalW * scale);
+        this.baseSize.height = Math.round(naturalH * scale);
+      },
+      onWindowResize() {
+        const img = this.$refs.imageRef;
+        if (img && img.complete) {
+          this.calculateSize();
+          this.clampDelta();
+        }
+      },
+      getDeltaLimits() {
+        const DeltaLimitX = Math.max(
+          (this.baseSize.width * this.scale - this.backdropSize.width) / 2,
+          0,
+        );
+        const DeltaLimitY = Math.max(
+          (this.baseSize.height * this.scale - this.backdropSize.height) / 2,
+          0,
+        );
+        return { DeltaLimitX, DeltaLimitY };
+      },
+      clampDelta() {
+        const { DeltaLimitX, DeltaLimitY } = this.getDeltaLimits();
+        this.delta.x = clamp(this.delta.x, -DeltaLimitX, DeltaLimitX);
+        this.delta.y = clamp(this.delta.y, -DeltaLimitY, DeltaLimitY);
+      },
+      onMouseDown(e) {
+        if (this.scale <= 1) return; // No reposition if not zoomed
+        e.preventDefault();
+        this.isDragging = true;
+        this.allowTransition = false; // disable transitions while dragging
+        this.dragStart = { x: e.clientX, y: e.clientY };
+        this.origin = { x: this.delta.x, y: this.delta.y };
+        window.addEventListener('mousemove', this.onMouseMove);
+        window.addEventListener('mouseup', this.onMouseUp);
+      },
+      onMouseMove(e) {
+        if (!this.isDragging) return;
+        this.delta.x = this.origin.x + (e.clientX - this.dragStart.x);
+        this.delta.y = this.origin.y + (e.clientY - this.dragStart.y);
+        this.clampDelta();
+      },
+      onMouseUp() {
+        this.isDragging = false;
+        this.allowTransition = true;
+        window.removeEventListener('mousemove', this.onMouseMove);
+        window.removeEventListener('mouseup', this.onMouseUp);
+      },
+      onKeyDown(e) {
+        this.handleArrowKeys(e);
+        this.handleTab(e);
+      },
+      handleArrowKeys(e) {
+        if (this.scale === 1) return;
+        const step = 50;
+        if (e.key === 'ArrowLeft') this.delta.x += step;
+        if (e.key === 'ArrowRight') this.delta.x -= step;
+        if (e.key === 'ArrowUp') this.delta.y += step;
+        if (e.key === 'ArrowDown') this.delta.y -= step;
+        this.clampDelta();
+      },
+      handleTab(e) {
+        var focusables = this.$refs.dialogRef.querySelectorAll('button:not([disabled])');
+        if (!focusables.length) return;
+        var firstFocusable = focusables[0];
```

Comment:
Let's use `const` by default: it makes your code safer and communicates intent.` var` is generally discouraged in modern JavaScript due to its less predictable scoping and hoisting behavior compared to let and const. https://www.freecodecamp.org/news/var-let-and-const-whats-the-difference/